pkg/subscriptions/subscriptions.js

```diff
@@ -409,76 +397,116 @@ define([
+            promise.cancel = function cancel() {
+                process.close("cancelled");
+            };
+
+            return promise;
+        }
+
+        function register_system(url, username, password) {
+            if (action_in_progress()) {
+                console.log(_('Unable to register at this time because a call to subscription manager ' +
+                  'is already in progress. Please try again.'));
+                return;
+            }
+            /* remove old errors */
+            remove_notifications();
+
+            system_unregistered.hide();
+            var registering_system = $('#subscriptions-registering');
+            registering_system.show();
+            var btn = $('#account-register-start');
+            btn.prop("disabled", true);
+
+            perform_register(url, username, password)
+                .always(function() {
                     registering_system.hide();
                     btn.prop("disabled", false);
-
+                })
+                .done(function() {
                     $('#subscriptions-register-dialog').modal('hide');
                     /* trigger update just in case */
                     update_subscriptions();
                 })
                 .fail(function(ex) {
-                    registering_system.hide();
-                    btn.prop("disabled", false);
-
-                    /* detect error types we recognize, fall back is generic error */
-                    var invalid_username_string = 'Invalid username or password.';
-                    var invalid_credentials_string = 'Invalid Credentials';
                     var details = $('#subscriptions-register-password-note-details');
-                    var remainder;
-                    if (buffer.indexOf(invalid_username_string) === 0) {
-                        $('#subscriptions-register-password-note').show();
-                        $('#subscription-register-username').parent().addClass('has-error');
-                        $('#subscription-register-password').parent().addClass('has-error');
-                        remainder = buffer.substring(invalid_username_string.length).trim();
-                        if (remainder !== '') {
-                            details.html(linkify_message(remainder));
-                            details.show();
-                        }
-                    } else if (buffer.indexOf(invalid_credentials_string) === 0) {
+                    switch(ex.reason) {
+                    case "credentials":
                         $('#subscriptions-register-password-note').show();
                         $('#subscription-register-username').parent().addClass('has-error');
                         $('#subscription-register-password').parent().addClass('has-error');
-                        /* if there is more to the message, show that */
-                        remainder = buffer.substring(invalid_credentials_string.length).trim();
```

Comment:
So we now always show the full message from the registration server? Is that an intended change?